zulipterminal/ui_tools/views.py

```diff
@@ -1687,3 +1687,49 @@ def keypress(self, size: urwid_Size, key: str) -> str:
             )
             return key
         return super().keypress(size, key)
+
+
+class FullRenderedMsgView(PopUpView):
+    def __init__(
+        self,
+        controller: Any,
+        message: Message,
+        topic_links: "OrderedDict[str, Tuple[str, int, bool]]",
+        message_links: "OrderedDict[str, Tuple[str, int, bool]]",
+        time_mentions: List[Tuple[str, str]],
+        title: str,
+    ) -> None:
+        self.controller = controller
+        self.message = message
+        self.topic_links = topic_links
+        self.message_links = message_links
+        self.time_mentions = time_mentions
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+
+        # Get rendered message
+        msg_box = MessageBox(message, controller.model, None)
+
+        super().__init__(
+            controller,
+            [msg_box.content],
+            "FULL_RENDERED_MESSAGE",
+            max_cols,
+            title,
+            urwid.Pile(msg_box.header),
+            urwid.Pile(msg_box.footer),
+        )
+
+    def keypress(self, size: urwid_Size, key: str) -> str:
+        if is_command_key("GO_BACK", key) or is_command_key(
+            "FULL_RENDERED_MESSAGE", key
+        ):
+            self.controller.show_msg_info(
+                msg=self.message,
+                topic_links=self.topic_links,
+                message_links=self.message_links,
+                time_mentions=self.time_mentions,
+            )
+            return key
+        elif is_command_key("MSG_INFO", key):
+            self.controller.exit_popup()
```

Comment:
I can see why this is necessary, but I'm unclear why we don't use it for the edit history popup, since that seems a close equivalent of this one.

There's no test for this?

Reply:
Ah. This didn't grab my attention before! I didn't expect this to be the case :sweat_smile:. Thanks for pointing this out! I am actually expecting the opposite. What I expect is exiting of popup using either `FULL_RENDERED_MESSAGE` (say) or `GO_BACK` **only**

Reply:
Good discussion! Thanks @preetmishra :)